### PR TITLE
Add  __traits(initSymbol) to the spec

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
The traits was upstreamed from LDC in dlang/dmd#13298

### CODE REVIEW

File: spec/traits.dd [hunk 1]
```diff
@@ -1713,6 +1714,39 @@ $(H2 $(GNAME toType))
 
         $(RATIONALE Provides the inverse operation of the $(DDSUBLINK property, mangleof, `.mangleof`) property.)
 
+$(H2 $(GNAME initSymbol))
+
+        $(P Takes a single argument, which must evaluate to a `class` or `struct` type.
+            Returns a `const(void)[]` that points to either the initializer symbol of the supplied
+            type or `null` if it is $(RELATIVE_LINK2 isZeroInit, zero-initialized). The referenced symbol
```
Comment (kinke) [inline spec/traits.dd:1721]: `__traits(isZeroInit, <class>)` is always true (=> class *ref*). All classes have init symbols (vptr is definitely not null).

Comment (MoonlightSentinel) [reply spec/traits.dd]: Right. Removed the link

File: spec/traits.dd [hunk 2]
```diff
@@ -1713,6 +1714,39 @@ $(H2 $(GNAME toType))
 
         $(RATIONALE Provides the inverse operation of the $(DDSUBLINK property, mangleof, `.mangleof`) property.)
 
+$(H2 $(GNAME initSymbol))
+
+        $(P Takes a single argument, which must evaluate to a `class` or `struct` type.
+            Returns a `const(void)[]` that points to either the initializer symbol of the supplied
+            type or `null` if it is zero-initialized. The referenced symbol holds the initial state
+            of any instance of the supplied type.
```
Comment (ljmf00) [inline spec/traits.dd:1722]: Can we refer to something about the length of the returned slice? From the example below, I understood that `__traits(initSymbol, C)` returns the correct length of the initializer memory even if zero-initialized and that matches the implementation, although this wording can be interpreted wrongly, IMO.

Comment (MoonlightSentinel) [reply spec/traits.dd]: Added a more explicit description for the `ptr` and `length` properties of  the returned slice.

**Review by ljmf00 — APPROVED** [review]
I don't know much about DDoc structure-wise, but it looks good to me

**Review by thewilsonator — APPROVED** [review]

**Pull request merged**